src/components/PostsList/PostsList.jsx

```diff
@@ -1,37 +1,87 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+
 import './PostsList.scss';
 
-export const PostsList = () => (
+export const PostsList = ({
+  posts,
+  selectedUser,
+  selectedPost,
+  setSelectedPost,
+  isOpen,
+  setIsOpen,
+  setAreCommentsHidden,
+}) => (
   <div className="PostsList">
     <h2>Posts:</h2>
 
     <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
+      {posts.map((post) => {
+        if (
+          selectedUser === '0'
+            || +selectedUser === post.userId
+        ) {
+          return (
+            <li
+              className="PostsList__item"
+              key={post.id}
+            >
+              <div>
+                <b>
+                  [User #
+                  {post.userId}
+                  ]:
+                  {' '}
+                </b>
+                {post.title}
+              </div>
+
+              <button
+                type="button"
+                className={`PostsList__button button ${
+                  (isOpen && +selectedPost.id === post.id)
+                    ? 'PostsList__close-button'
+                    : ''
+                }`}
 
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
+                onClick={() => {
+                  if (selectedPost.id !== post.id) {
+                    setSelectedPost(post);
+                    setIsOpen(true);
+                    setAreCommentsHidden(true);
+                  } else {
+                    setIsOpen(state => !state);
```

Comment:
you passing callback here, and couple strings above you passing a boolean param, that's not ok, type of state property should never change